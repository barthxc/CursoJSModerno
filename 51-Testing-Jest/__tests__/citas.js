import Citas from '../js/classes/Citas';

describe('Probar la clase de Citas', ()=>{

    const citas = new Citas();

    const id = Date.now()
    test('Agregar una nueva cita',()=>{



        const citaObj = {
            id,
            mascota: 'Duke',
            propietario: 'Pablo',
            telefono: '123123',
            fecha: '23-02-1996',
            hora:'14:00',
            sintomas: 'Se murió'
        }

        citas.agregarCita(citaObj);

        //PRUEBA
        expect(citas).toMatchSnapshot();
    });


    test('Actualizar cita', ()=>{
        const citaActualizada = {
            id,
            mascota: 'Nebula',
            propietario: 'Pablo',
            telefono: '123123',
            fecha: '23-02-1996',
            hora:'14:00',
            sintomas: 'Se murió'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });


    test('Eliminar cita', ()=>{
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    })
})