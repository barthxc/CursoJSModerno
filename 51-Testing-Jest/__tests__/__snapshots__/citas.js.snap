// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "23-02-1996",
      "hora": "14:00",
      "id": 1694079055680,
      "mascota": "Nebula",
      "propietario": "Pablo",
      "sintomas": "Se murió",
      "telefono": "123123",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "23-02-1996",
      "hora": "14:00",
      "id": 1694079055680,
      "mascota": "Duke",
      "propietario": "Pablo",
      "sintomas": "Se murió",
      "telefono": "123123",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
